py/qstr.c

```diff
@@ -192,12 +193,17 @@ qstr qstr_from_str(const char *str) {
 }
 
 qstr qstr_from_strn(const char *str, size_t len) {
-    assert(len < (1 << (8 * MICROPY_QSTR_BYTES_IN_LEN)));
     QSTR_ENTER();
     qstr q = qstr_find_strn(str, len);
     if (q == 0) {
         // qstr does not exist in interned pool so need to add it
 
+        // check that len is not too big
+        if (len >= (1 << (8 * MICROPY_QSTR_BYTES_IN_LEN))) {
+            QSTR_EXIT();
+            mp_raise_msg(&mp_type_SyntaxError, "name too long");
```

Comment:
How about using RuntimeError?  This error may now be raised from many locations, not just during parsing/compilation, so would be confusing if it were a SyntaxError that was raised during a function call (for example).  I think RuntimeError is quite appropriate because it's really a problem with the runtime not being able to handle the qstr.

Reply:
I agree with you! And I pushed the change.